dist-repo-builder/src/main/java/io/ballerina/dist/DistRepoBuilder.java

```diff
@@ -41,70 +49,107 @@
-                if (pathMatcher.matches(path)) {
-                    Path relativePath = jBalToolsDocPath.relativize(path.getParent());
-                    if (!relativePath.toString().equals("")) {
-                        existingDocs.add(jBalToolsDocPath.relativize(path.getParent()));
+    private static void generateCaches(List<Path> balas, Path repo) {
+        String[] stdLibs = {
+                //Standard Libraries
+                "ballerina-io",
+                "ballerinax-java.arrays",
+                "ballerina-random",
+                "ballerina-regex",
+                "ballerina-runtime",
+                "ballerina-task",
+                "ballerina-time",
+                "ballerina-url",
+                "ballerina-xmldata",
+                "ballerina-cache",
+                "ballerina-crypto",
+                "ballerina-os",
+                "ballerina-xslt",
+                "ballerina-log",
+                "ballerina-reflect",
+                "ballerina-uuid",
+                "ballerina-auth",
+                "ballerina-file",
+                "ballerina-ftp",
+                "ballerina-jwt",
+                "ballerina-mime",
+                "ballerina-nats",
+                "ballerina-oauth2",
+                "ballerina-stan",
+                "ballerina-tcp",
+                "ballerina-udp",
+                "ballerina-email",
+                "ballerina-http",
+                "ballerina-grpc",
+                "ballerinai-transaction",
+                "ballerina-websocket",
+                "ballerina-websub",
+                "ballerina-websubhub",
+                "ballerina-kafka",
+                "ballerina-rabbitmq",
+                "ballerina-sql",
+                "ballerinax-java.jdbc",
+                "ballerinax-mysql",
+                "ballerina-graphql",
+                "ballerina-graphql", // TODO : graphql had to generate twice
+                //Extensions
+                //TODO: Revisit extensions
+                "ballerina-openapi",
+                "ballerina-docker",
+                "ballerinax-awslambda",
+                "ballerinax-azure_functions",
+                "ballerinax-choreo",
+                "ballerinax-jaeger",
+                "ballerinax-prometheus",
+                "ballerinai-observe"
+        };
+
+        for (String stdLib : stdLibs) {
+            for (Path bala : balas) {
+                if (bala.getFileName().toString().contains(stdLib)) {
+                    System.out.println("Generating " + stdLib);
+                    try {
+                        System.setProperty(BALLERINA_HOME_KEY, repo.getParent().toString());
+                        ProjectEnvironmentBuilder defaultBuilder = ProjectEnvironmentBuilder.getDefaultBuilder();
+                        defaultBuilder.addCompilationCacheFactory(new FileSystemCache.FileSystemCacheFactory(repo));
```

Comment:
The path of cache should be `repo.resolve("cache")`

Reply:
Once this is fixed, remove L72-L73 (the second call to `generateCache()`) and see if this is working